docs/docs/data-structures/tree/transactions.mdx

```diff
@@ -0,0 +1,34 @@
+---
+title: Transactions
+sidebar_position: 6
+---
+
+If you want the `SharedTree` to treat a set of changes atomically, wrap these changes in a transaction.
+Using a transaction guarantees that (if applied) all of the changes will be applied together synchronously (though, note that the Fluid Framework guarantees this already for any sequence of changes that are submitted synchronously). However, the changes may not be applied at all if the transaction is given one or more constraints.
+If any constraint on a transaction is not met, then the transaction and all its changes will ignored by all clients.
+Additionally, all changes in a transaction will be reverted together as a single unit by [undo/redo code](./undo-redo.mdx), because changes within a transaction are exposed through a single `Revertible` object.
+It is also more efficient for SharedTree to process a large number of changes in a row as a transaction rather than as changes submitted separately.
+
+To create a transaction use the `Tree.runTransaction()` method. You can cancel a transaction from within the callback function by returning the special "rollback object", available via `Tree.runTransaction.rollback`. Also, if an error occurs within the callback, the transaction will be canceled automatically before propagating the error.
```

Comment:
Note that there is a new API for transaction - `[TreeViewAlpha:runTransaction](https://github.com/microsoft/FluidFramework/blob/9bdb9b5d606d9312f3679cf501edff5c8dd68537/packages/dds/tree/src/simple-tree/api/tree.ts#L581)`.

So, this document needs to be updated. Maybe in a follow-up PR.